examples/random-login/main.go

```diff
@@ -252,5 +269,29 @@ func (r *RandomSalt) WireDependencies(f infer.FieldSelector, args *RandomSaltArg
 }
 
 type Config struct {
-	Scream *bool `pulumi:"itsasecret,optional"`
+	Scream *bool `pulumi:"scream,optional"`
+}
+
+type UsernameIsUnique struct{}
+
+func (UsernameIsUnique) Invoke(ctx context.Context, req infer.FunctionRequest[UsernameIsUniqueArgs]) (infer.FunctionResponse[UsernameIsUniqueRes], error) {
+	return infer.FunctionResponse[UsernameIsUniqueRes]{
+		// Here we would call the database to check if the username already exists.
+		// For now we will always return true, but we will mock different return values in the tests.
+		Output: UsernameIsUniqueRes{
+			IsUnique: true,
+		},
```

Comment:
Consider simplifying this function to return a boolean rather than `UsernameIsUniqueRes`.

Reply:
I think I prefer it using the type rather than the boolean? For pattern's sake